### PR TITLE
test(build/testing): add integration case for local fileystem

### PR DESCRIPTION
**Pull request by GeorgeMac**
This adds a failing integration test case for the `local` filesystem backend.

Currently, the attempts fail to start Flipt because the `local` filesystem hasn't been implemented.

However, once implemented, the test configures Flipt over the seed data in the readonly test suite.
Then it runs the readonly suite against Flipt. This should ensure compatibility with import and export behaviour.

I haven't attempted the `git` filesystem in this PR, as that takes more work to configure.
That can be done in a subsequent PR (I think we could use something like Gitea).

### CODE REVIEW

File: build/testing/integration.go [hunk 1]
```diff
@@ -21,6 +21,12 @@ var (
 	protocolPorts = map[string]string{"http": "8080", "grpc": "9000"}
 	replacer      = strings.NewReplacer(" ", "-", "/", "-")
 	sema          = make(chan struct{}, 6)
+
+	defaultCases = map[string]testCaseFn{
+		"api":           api,
+		"fs/local":      local,
+		"import/export": importExport,
+	}
```
Comment (yquansah) [inline build/testing/integration.go:29]: I wonder if we should call this variable `allCases` instead. I feel like "default" has a different meaning connotatively.

File: build/testing/integration.go [hunk 2]
```diff
@@ -30,27 +36,57 @@ type testConfig struct {
...
+
+	cases := map[string]testCaseFn{}
+	for n, fn := range defaultCases {
+		for _, filter := range caseNames {
+			if n == filter {
+				cases[n] = fn
+			}
+		}
+	}
```
Comment (yquansah) [inline build/testing/integration.go:59]:
```suggestion
	for _, filter := range caseNames {
	      cases[filter] = defaultCases[filter]
	}
```

Would something like this work?

File: build/testing/integration.go [hunk 3]
```diff
@@ -182,16 +227,15 @@ func importExport(ctx context.Context, base, flipt *dagger.Container, conf testC
 		namespace := conf.namespace
 		if namespace == "" {
 			namespace = "default"
+			// replace namespace in expected yaml
+			expected = strings.ReplaceAll(expected, "version: \"1.0\"\n", fmt.Sprintf("version: \"1.0\"\nnamespace: %s\n", namespace))
```
Comment (yquansah) [inline build/testing/integration.go:231]: Nice I was wondering how the `export` functionality worked as far as assertions, when data was imported with no `namespace`.

**Review by yquansah — COMMENTED** [review]
Just a couple comments. But awesome work.

**Review by yquansah — COMMENTED** [review]
Just a couple comments. But awesome work.

File: build/testing/integration.go (hunk 2 above)
Comment (GeorgeMac) [reply build/testing/integration.go:59]: Very much so. Great catch! 🙇

File: build/testing/integration.go (hunk 3 above)
Comment (GeorgeMac) [reply build/testing/integration.go:231]: Yeah it is a bit fiddly handling these edge cases.

**Comment by GeorgeMac** [comment]
This has been superseded.

**Pull request closed without merging**